apps/omg_watcher_info/lib/omg_watcher_info/api/stats.ex

```diff
@@ -0,0 +1,68 @@
+# Copyright 2019-2020 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.WatcherInfo.API.Stats do
+  @moduledoc """
+  Module provides operations related to network statistics.
+  """
+
+  alias OMG.WatcherInfo.DB.Block
+  alias OMG.WatcherInfo.DB.Transaction
+
+  @doc """
+  Retrieves network statistics.
+  """
+  def get() do
+    timestamps_all_time = Block.get_timestamps()
+    timestamps_last_24_hours = Block.get_timestamps_last_24_hours()
+
+    response = %{
+      transaction_count: %{
+        all_time: Transaction.get_count(),
+        last_24_hours: Transaction.get_count_last_24_hour()
+      },
+      block_count: %{
+        all_time: Block.get_count(),
+        last_24_hours: Block.get_count_last_24_hour()
+      },
+      average_block_interval: %{
+        all_time: get_average_block_interval(timestamps_all_time),
+        last_24_hours: get_average_block_interval(timestamps_last_24_hours)
+      }
+    }
+
+    {:ok, response}
+  end
+
+  @spec get_average_block_interval([%{timestamp: integer}]) :: float | String.t()
+  def get_average_block_interval(timestamps) do
+    case length(timestamps) do
```

Comment:
http://erlang.org/doc/efficiency_guide/commoncaveats.html#length-1

Would be better to do...

```elixir
case timestamps do
  [_, _, | _] ->
    # calculate

  _ ->
    # skip
end
```

The reason I'm suggesting this here is because this function could be calculating for all blocks, not limited by a specific period.

Reply:
However, before making changes to this. Please see https://github.com/omisego/elixir-omg/pull/1273/files#r369961022

Reply:
Thanks for the syntactical (and time complexity) improvement! Done 61afded.

Reply:
https://github.com/omisego/elixir-omg/pull/1273/files/98fbd15582f2245d4b7c9095448ad87a66cae5aa..8946c62c598f52b5dbe7f2b718510119ded86b25#r370544428